### PR TITLE
Teach cmake how to find the installed PyTorch.

### PR DESCRIPTION
**Pull request by stellaraccident**
* In most situations, this eliminates the need to explicitly set a path to the Torch cmake files.
* Also upgrades to new Python3 find package. (should eliminate 2.x mismatches)
* Since PyTorch is located by asking Python where it is, this eliminates a lot of causes of mismatch. (one source of truth)

### CODE REVIEW

**Review by silvasean — APPROVED** [review]

**Comment by silvasean** [comment]
IREE changes LG. It always makes me happy when we can live with binary dependencies :)

**Comment by stellaraccident** [comment]
> IREE changes LG. It always makes me happy when we can live with binary dependencies :)

Oh, whoops - messed up my branching and didn't mean to push that in this pr. Fine though to do it :)

File: cmake/modules/ConfigurePyTorch.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  else()
+    message(STATUS "Checking for PyTorch using ${PYTHON_EXECUTABLE} ...")
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+      -c "import os;import torch;print(os.path.dirname(torch.__file__), end='')"
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      RESULT_VARIABLE PYTORCH_STATUS
+      OUTPUT_VARIABLE PYTORCH_PACKAGE_DIR)
+    if(NOT PYTORCH_STATUS EQUAL "0")
+      message(FATAL_ERROR "Unable to 'import torch' with ${PYTHON_EXECUTABLE}")
```
Comment (mikeurbach) [inline cmake/modules/ConfigurePyTorch.cmake:13]: When I tried running `./build_tools/cmake_configure.sh` without PyTorch installed, I did hit this, and it exited the script. If that is the intent, should this also be checking if PyTorch is required? I was using the default setting where PyTorch is optional, and I expected to be able to proceed without it.

**Review by mikeurbach — COMMENTED** [review]
I was literally just building mlir-npcomp for the first time, so I gave this a whirl before and after doing a `pip install`. See my comment below about some confusion when PyTorch is not installed. After installing PyTorch, this successfully found it for me :+1:

File: cmake/modules/ConfigurePyTorch.cmake (hunk 1 above)
Comment (stellaraccident) [reply cmake/modules/ConfigurePyTorch.cmake:13]: Yeah, good catch. The CI also complained to me about that and what is now landed at head does the right thing when PyTorch isn't installed. I just had a git problem and didn't update the PR properly before pushing.

**Pull request closed without merging**